### ISSUE TITLE
Masked card revealed to non-owner

### ISSUE THREAD

**Issue by riverwanderer** [vassalengine/vassal#12125]
Problem: In a game between two players, "French" player set a face-down card aside onto the board (an automated action). Normally, this card is Obscured to the other player. In this case however, that player ("Allies") was able to see the masked card.
In other parts of the game, other masked cards performed as expected for both sides.

Module: [C&C Napoleonics v4.1.7](https://vassalengine.org/wiki/File:CCNapoleonics4_1_7.vmod)

Log: [Jay and Paul CCN.zip](https://github.com/vassalengine/vassal/files/11023713/Jay.and.Paul.CCN.zip)

Incident occurs when a card from Paul's hand is moved to the French "Square row" - bottom right of main board. 

Jay correctly reported the actual card that Paul had set aside, despite this card showing as "Hidden" i.e. Masked.

Both players were using Vassal v4.6.14 and had unique passwords. All other aspects of card handling / masking in the module performed as expected.

The module is set to never allow other players to Unmask cards.

The card type has the following simple Mask trait:
![image](https://user-images.githubusercontent.com/58135975/226484169-f35ccd9c-2d9f-42e1-be47-add93b8ad2fe.png)

**Comment by Cattlesquat** [comment]
The "Square row" isn't a deck by any chance, is it?

**Comment by riverwanderer** [comment]
> The "Square row" isn't a deck by any chance, is it?
No. It is a set of zones - 1 per card slot, each containing an Irregular Grid with 1 unique region.

**Further info:**
The "squaring action" moves the card (_send to location_ to a Region) from a deck hidden (-ve offset) in the hand window of the side that owns the card.

In this game:
i) The first card moved to a Square slot during the game was by the Allies side (top right of board). This card was properly masked against the French side. 
ii) Other cards drawn to the board during this game by a different action were correctly masked against either side.
iii) It is possible to connect into the Logfile as Allies (Allies side was free at start of logfile). When you do so, the subject card is corrected masked against you. It was definately not masked against the Allies player in the game ("Jay"), as he was able to identify the card. The log file shows that the card was not moved or manipulated by either player whilst on the board.

**Comment by riverwanderer** [comment]
Also, there is something unusual about the start up sequence for this game. After Paul draws his has hand, the game clock is enabled, which is odd (only explanation I can think of is that Jay joined the game and enabled that whilst out of sync with Paul). 

Secondly, if you UnDo from where Paul has drawn his 5 cards, the game start setup is restored as expected. However, if you procede to the step after ("game clocks enabled" messages) and then attempt to UnDo, the game state gets corrupted once you UnDo that step. I think this confirms that Jay was in the middle of joining the game when Paul was starting the game.

However, all this may have nothing to do with the Square card issue later.
